src/main/java/org/fundacionjala/coding/rfalconi/EAN.java

```diff
@@ -0,0 +1,52 @@
+package org.fundacionjala.coding.rfalconi;
+
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+public class EAN {
+
+    private static final Logger LOGGER = Logger.getLogger("org.fundacionjala.coding.rfalconi.EAN");
+    private static final int TEN = 10;
+
+    public String getChecksum(final String barcode) {
```

Comment:
Your implementation has 3 lops (for and while)
You can solve this problem with a single for loop

Reply:
Solution implemented using a for loop with if

Reply:
Now you have a good approach
But you should return true in the validate method.
Take as examples the following tests.
```java
    /**
     * First Code "9783815820865".
     */
    @Test
    public void test01() {
        assertTrue("Check EAN-Code \"9783815820865\"", EanValidator.validate("9783815820865"));
    }

    /**
     * Second code "9783815820864".
     */
    @Test
    public void test02() {
        assertFalse("Check EAN-Code \"9783815820864\"", EanValidator.validate("9783815820864"));
    }

    /**
     * Third code "9783827317100".
     */
    @Test
    public void test03() {
        assertTrue("Check EAN-Code \"9783827317100\"", EanValidator.validate("9783827317100"));
    }
```